### PR TITLE
Admin can visit dashboard

### PR DESCRIPTION
**Pull request by janderson16**
-created tests/ functionality for admin
-created admin dashboard and corresponding controller
-fixed @maxglassie 's subpar Cart-Total method

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -11,16 +11,8 @@ def current_user
...
-    render file: "/public/404" unless current_user && current_admin?
-  end
-
   def current_admin?
-    if current_user.role == 1
-      true
-    else
-      false
-    end
+    current_user && current_user.admin?
```
Comment (janderson16) [inline app/controllers/application_controller.rb:15]: refactored

File: app/models/cart.rb [hunk 2]
```diff
@@ -35,7 +35,8 @@ def total
     @cart_items.each do |item, quantity|
       sum += (item.price * quantity)
     end
-    sum
+    sum.round(2)
```
Comment (janderson16) [inline app/models/cart.rb:38]: left cart total in .66666666666666666666666666666

File: spec/features/cart/user_can_add_items_to_cart_spec.rb [hunk 3]
```diff
@@ -19,8 +19,8 @@
 
         expect(page).to have_content "Item: Lion Holographic Slammer"
         expect(page).to have_content "Quantity: 1"
-        within("div.subtotal") do
-          expect(page).to have_content "Subtotal: $3.29"
+        within("div.total") do
```
Comment (janderson16) [inline spec/features/cart/user_can_add_items_to_cart_spec.rb:22]: fixed test to match view

File: app/models/cart.rb (hunk 2 above)
Comment (maxglassie) [reply app/models/cart.rb:38]: nice update there. good to keep things trimmed.

File: config/routes.rb [hunk 4]
```diff
@@ -5,8 +5,11 @@
   post '/login', to: "sessions#create"
   delete '/logout', to: "sessions#destroy"
 
+  namespace :admin do
+    get 'dashboard', to: "dashboard#dashboard"
```
Comment (maxglassie) [inline config/routes.rb:9]: nice work on this custom method to send to the dashboard controller for admins.

File: spec/features/cart/user_can_add_items_to_cart_spec.rb (hunk 3 above)
Comment (maxglassie) [reply spec/features/cart/user_can_add_items_to_cart_spec.rb:22]: thanks for cleaning this up.

**Comment by maxglassie** [comment]
This looks good. Nice job figuring out that custom route for the admin controller. The tests look good to make sure that login functionality works properly -- I especially like those model tests. Also, thanks for cleaning up and fixing the little details in the cart.

**Pull request merged**